### PR TITLE
{monitor}`az monitor clone`: Migrate command to codegen V2

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

az monior clone
**Description**

**Testing Guide**

**History Notes**

{monitor}`az monitor clone`: Migrated to codegen V2

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
monitor

File: src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py [hunk 1]
```diff
@@ -2,98 +2,120 @@
...
+            }
+            alert_rule['actions'][index] = new_action
+            action_group_mapping[action['action_group_id']] = [new_action_group['id'], new_action]
     return alert_rule
 
 
-def _clone_alert_rule(monitor_client, alert_rule, target_resource):
-    alert_rule.scopes = [target_resource]
+def format_metrics_alert_req(alert_rule):
+    all_of = alert_rule["criteria"]["all_of"]
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py:74]: Do you need to pop the "all_of"?

File: src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py [hunk 2]
```diff
@@ -2,98 +2,120 @@
...
+
+
+def _add_into_existing_scopes(cmd, source_resource, alert_rule, target_resource):
+    subscription_id = parse_resource_id(source_resource)['subscription']
+    alert_rule['scopes'].append(target_resource)
+    resource_group_name, name = _parse_id(alert_rule['id']).values()  # pylint: disable=unbalanced-dict-unpacking
+    alert_rule["subscription"] = subscription_id
+    alert_rule["resource_group"] = resource_group_name
+    alert_rule["name"] = name
+    from ..aaz.latest.monitor.metrics.alert import Update
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py:38]: Is there need `format_metrics_alert_req(alert_rule)` as well?

Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py:38]: Only `scopes` are updated so no need to format other parameters, they are useless in `Update` cmd

File: src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py (hunk 1 above)
Comment (AllyW) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py:74]: `all_of` for `criteria` will be override in the following operations.

File: src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py (hunk 2 above)
Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py:38]: In that case, please specify the arg names passed into command_args. Such as 
```
command_args={
  "name": name,
  "resource_group": resource_group,
  "subscription": subscription_id,
  "scopes": scopes,
}
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**